File: server/medicore.layer/MediCore.Service/Model/MedicalStaffSpecialistMapModel.cs
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class MedicalStaffSpecialistMapModel : IModelBase
    {
        public int Id { get; set; }
        public int MedicalStaffId { get; set; }
        public int MedicalStaffSpecialistId { get; set; }

       
        public virtual MedicalStaffModel MedicalStaff { get; set; }
       
        public virtual MedicalStaffSpecialistModel MedicalStaffSpecialist { get; set; }
    }
}

File: server/medicore.apps/src/model/medicalStaffSpecialistMapModel.ts
import { Type } from 'class-transformer';
import { IEntityBase } from './IEntityBase';
import { MedicalStaffModel } from './medicalStaffModel';
import { MedicalStaffSpecialistModel } from './medicalStaffSpecialistModel'

export class MedicalStaffSpecialistMapModel implements IEntityBase {
    id: number;
    medicalStaffId: number;
    medicalStaffSpecialistId: number;

    @Type(() => MedicalStaffModel)
    medicalStaff: MedicalStaffModel;
    @Type(() => MedicalStaffSpecialistModel)
    medicalStaffSpecialist: MedicalStaffSpecialistModel;
}